pytest_plugins/rerun_rp/rerun_rp.py

```diff
@@ -56,48 +85,63 @@ def pytest_addoption(parser):
+    help_text = '''
+            Collects user tests from report portal.
+
+            Usage: --user [value]
+
+            Value: [ Report_portal_username ]
+
+            Defect_types:
+                Collect tests of specific user from Report Portal
+        '''
+    parser.addoption("--user", nargs='?', help=help_text)
 
 
 @pytest.hookimpl(tryfirst=True)
 def pytest_collection_modifyitems(items, config):
     """
-    Collects tests based on pytest option to select tests marked as failed on Report Portal
+    Collects and modifies tests collection based on pytest options to select tests marked as
+    failed/skipped and user specific tests in Report Portal
     """
     fail_args = config.getoption('only_failed', False)
-    if not fail_args:
+    skip_arg = config.getoption('only_skipped', False)
+    user_arg = config.getoption('user', False)
+    if not any([fail_args, skip_arg, user_arg]):
         return
     rp = ReportPortal()
-    fail_args = fail_args.split(',') if ',' in fail_args else [fail_args]
-    allowed_args = [*rp.defect_types.keys()]
-    # Stop Session based on Arguments to pytest
-    if not fail_args == ['all']:
-        if not set(fail_args).issubset(set(allowed_args)):
-            raise pytest.UsageError(
-                f'Incorrect values to pytest option \'--only-failed\' are provided as '
-                f'{fail_args} but should be none/one/mix of {allowed_args}'
-            )
-    # Fetch the latest launch
     version = settings.server.version
     sat_version = f'{version.base_version}.{version.epoch}'
     launch = next(iter(rp.launches(sat_version=sat_version).values()))
-    # Stop session based on RP Launch statistics and info
-    if launch.info['isProcessing']:
-        raise LaunchError(f'The launch of satellite version {sat_version} is not Finished yet')
-    fail_percent = round(
-        (int(launch.statistics['failed']) / int(launch.statistics['total']) * 100)
-    )
-    if fail_percent > 20:
-        raise LaunchError(
-            f'The latest launch of Satellite verson {sat_version} has {fail_percent}% tests '
-            'failed. Which is higher than the threshold of 20%. '
-            'Examine the failures thoroughly and check for any major issue.'
-        )
-    rp_tests = _get_failed_tests(launch, fail_args)
+    _validate_launch(launch, sat_version)
+    test_args = {}
+    if fail_args:
+        test_args['status'] = 'failed'
+        if not fail_args == ['all']:
+            defect_types = fail_args.split(',') if ',' in fail_args else [fail_args]
+            allowed_args = [*rp.defect_types.keys()]
+            if not set(defect_types).issubset(set(allowed_args)):
+                raise pytest.UsageError(
+                    'Incorrect values to pytest option \'--only-failed\' are provided as '
+                    f'\'{fail_args}\'. It should be none/one/mix of {allowed_args}'
+                )
+            test_args['defect_types'] = defect_types
+    elif skip_arg:
```

Comment:
I think it would be best for this to be additive, so that one could run `pytest --only-failed all --only-skipped` to run all skipped and failed tests from the launch.

Reply:
yeah, earlier I thought of that but just wanted to keep them separate ! 

Do you agree if I raise a separate PR to combine them as this PR is pending from long ot its altogether an improvement.